src/main/java/io/jsonwebtoken/JwtParts.java

```diff
@@ -0,0 +1,72 @@
+package io.jsonwebtoken;
+
+/**
+ * Container class which contains encoded header, payload and signature parts of
+ * the token.
+ *
+ * @since 0.8
+ */
+public class JwtParts {
```

Comment:
This is not entirely correct.  A JWT can be a JWS or JWE, and they have different numbers of 'parts' (Unsecured JWT = 2 parts, JWS = 3, JWE = 5).  Whatever implementation we'd have around that needs to account for any/all 3 of these scenarios, and probably typed accordingly, e.g. `JwsParts extends JwtParts` and `JweParts extends JwtParts`.